rir/src/compiler/translations/rir_2_pir/rir_2_pir.cpp

```diff
@@ -281,61 +277,61 @@ bool Rir2Pir::compileBC(const BC& bc, Opcode* pos, Opcode* nextPos,
             }
         };
         auto ldfun = LdFun::Cast(callee);
-        if (monomorphic && isValidClosureSEXP(monomorphic)) {
-            // Currently we can only use StaticCall if we have exactly the right
-            // number of arguments.
-            // TODO, support cases where we need to pass Missing::instance() to
-            // pad the missing arguments. This is a bit tricky, because
-            // currently PIR assumes that no arguments are missing.
-
-            bool correctOrder =
-                (bc.bc == Opcode::named_call_implicit_)
-                    ? ArgumentMatcher::reorder(FORMALS(monomorphic),
-                                               bc.callExtra().callArgumentNames,
-                                               args)
-                    : RList(FORMALS(monomorphic)).length() == args.size();
-
-            if (!correctOrder) {
+        if (monomorphic) {
+            auto target = callTargetFeedback.at(callee).getTarget(srcCode, 0);
+
+            auto insertSpecPrologue = [&]() {
+                Value* expected = insert(new LdConst(target));
```

Comment:
but this can't work. this guard will just immediately fail. you are still comparing with the closure here, which, as you note in this pr, changes on every invocation.